### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by moise-mulungu**
On this milestone, I:
- Created a class named `nameable`
- Implemented a `correct_name` method that raised `NotImplementedError` for the `nameable`
- Made the class `Person` inherit  from the `Nameable` class
- Prepared three decorators; the `base,` `capitalize` and `trimmer`
- Updated the README file

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,5 +1,8 @@
-class Person
+require './nameable'
+
+class Person < Nameable
   def initialize(age, name = 'Unknown', parent_permission: true)
```
Comment (OLIPLICHE) [inline person.rb:5]: - [ ] Please set `parent_permission = true` instead of `parent_permission: true`

File: decorators.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+require './nameable'
+
+# Prepare a base decorator class
+class BaseDecorator
```
Comment (OLIPLICHE) [inline decorators.rb]: - [ ] Please make sure `BaseDecorator` class inherits from Namable. Hint: `class BaseDecorator < Nameable`

File: decorators.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+class CapitalizeDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+# Prepare a trimmer decorator class
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    @nameable.correct_name[0..10]
```
Comment (OLIPLICHE) [inline decorators.rb]: - [ ] Please make sure The name has a maximum of 10 characters. If it's longer it should trim the word. This is the requirement from the Project

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @moise-mulungu , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

Linter is green (the last commit have a green check mark (✅)) 👏
README file is descriptive 👏
Commit messages are meaningful 👏 👏
Github flow is correctly used (Pull Request is made from the feature branch into the main branch) 👏
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (moise-mulungu) [reply person.rb:5]: Hello @OLIPLICHE, thank you for your review. But from line 4 of the code, the equal operator throws an error when running `robocop`. I then kept the `colon` for that particular parameter

Comment (OLIPLICHE) [reply person.rb:5]: Hi, Sorry I am not in front of my machine and I am using the phone to reply. Kindly discuss about it with your team members. then submit it, the second reviewer will check.

Comment (moise-mulungu) [reply person.rb:5]: I did that, all of them used `colon` instead of the `equal operator` as you are suggesting.

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @moise-mulungu, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**